src/ORM/Table.php

```diff
@@ -2581,6 +2581,8 @@ public function hasFinder(string $type): bool
      */
     public function callFinder(string $type, Query $query, array $options = []): Query
     {
+        assert(empty($options) || !array_is_list($options), 'Finder options should be an associative array not a list');
```

Comment:
This isn't going to prevent numeric keys when mixed with string keys.

Reply:
That's a good point. I wasn't sure I would clean this up any better for 5.0, so it seems good enough until then.